test/utils/task.go

```diff
@@ -0,0 +1,88 @@
+package utils
+
+import (
+	"context"
+
+	"github.com/gardener/etcd-druid/internal/task"
+
+	"github.com/go-logr/logr"
+	"k8s.io/apimachinery/pkg/types"
+)
+
+// Ensure the fake fulfils the interface.
+var _ task.Handler = (*FakeHandler)(nil)
+
+// FakeHandler returns the same result every time its methods are called.
+// *Not* safe for concurrent mutation, but fine for typical single‑threaded
+// unit‑test flows.
+//
+// Each field has a sensible default (Completed=true) so tests can often create
+// one with `NewFakeHandler(...)` and be done.
+//
+// If you need different behaviour set the fields directly or use the fluent
+// helpers (WithAdmit, WithRun, WithCleanup).
+
+type FakeHandler struct {
+	etcdRef types.NamespacedName
+	name    string
+	logger  logr.Logger
+
+	AdmitResult   *task.Result
+	RunResult     *task.Result
+	CleanupResult *task.Result
+}
+
+// NewFakeHandler returns a fake whose Admit, Run, and Cleanup all immediately
+// signal completion.  You can override any of the results via the fluent
+// helper methods below.
+func NewFakeHandler(name string, etcdRef types.NamespacedName, logger logr.Logger) *FakeHandler {
+	completed := &task.Result{Completed: true}
```

Comment:
In the previous comment's conext, this will need to be changed to pass the correct expected state to each result type.